plugins/scaffolder/src/extensions/index.tsx

```diff
@@ -25,9 +25,13 @@ import { Extension, attachComponentData } from '@backstage/core-plugin-api';
 export const FIELD_EXTENSION_WRAPPER_KEY = 'scaffolder.extensions.wrapper.v1';
 export const FIELD_EXTENSION_KEY = 'scaffolder.extensions.field.v1';
 
-export function createScaffolderFieldExtension<T = any>(
-  options: FieldExtensionOptions<T>,
-): Extension<() => null> {
+export function createScaffolderFieldExtension<
+  TReturnValue = unknown,
+  TInputProps = unknown,
+>(
+  options: FieldExtensionOptions<TReturnValue, TInputProps>,
+  // TODO: need know how to embed these types nicely so the api report looks nice.
```

Comment:
@Rugvip, i need some of your typescript foo here. I want to embed the `InputProps` type so that we can see it in the api-report, not sure of the best way to do this. This I've found works, but it's ugly and probably not the best way.

It looks cleaner: 

https://github.com/backstage/backstage/pull/9752/files#diff-3091488929fe959b0e9a5090ac4eba13d26d0e4b6b89c70fe10e555e7a2f25bcR54 

but wondered if you had any better suggestion here as to make the types better.